php-checks/src/main/java/org/sonar/php/checks/CommentedOutCodeCheck.java

```diff
@@ -42,84 +40,98 @@
 
-  private final CodeRecognizer codeRecognizer = new CodeRecognizer(THRESHOLD, new PHPRecognizer());
-  private final Pattern regexpToDivideStringByLine = Pattern.compile("(\r?\n)|(\r)");
+  private static final String INNER_METHOD_SYNTAX_FORMAT = "class DummyClass{public function dummyMethod(){%s}}";
+  private static final String INNER_CLASS_SYNTAX_FORMAT = "class DummyClass{%s}";
 
-  private SyntaxTrivia previousTrivia;
-
-  private static class PHPRecognizer implements LanguageFootprint {
-
-    @Override
-    public Set<Detector> getDetectors() {
-      return ImmutableSet.of(
-        new EndWithDetector(0.95, '}', ';', '{'),
-        new KeywordsDetector(0.3, PHPKeyword.getKeywordValues()),
-        new ContainsDetector(0.95, "*=", "/=", "%=", "+=", "-=", "<<=", ">>=", "&=", "^=", "|="),
-        new ContainsDetector(0.95, "!=", "!=="));
-    }
-
-  }
+  private static final ActionParser<Tree> PARSER = PHPParserBuilder.createParser(PHPLexicalGrammar.TOP_STATEMENT);
+  private static final Deque<SyntaxTrivia> singleLineTrivias = new ArrayDeque<>();
 
   @Override
   public void visitCompilationUnit(CompilationUnitTree tree) {
-    previousTrivia = null;
     super.visitCompilationUnit(tree);
+
+    // check possible last single line comment and clean up afterwards
+    checkSingleLineComments();
   }
 
   @Override
   public void visitTrivia(SyntaxTrivia trivia) {
-    super.visitTrivia(trivia);
-
-    if (isInlineComment(trivia)) {
+    String comment = trivia.text();
+    if (comment.startsWith("/*") && isParsableCode(comment.replaceAll(MULTILINE_COMMENT_REPLACE, " "))) {
+      context().newIssue(this, trivia, MESSAGE);
+    }
+    if (comment.startsWith("//") || comment.startsWith("#")) {
+      collectSingleLineComment(trivia);
+    }
 
-      if (isCommentedCode(getContents(trivia.text())) && !previousLineIsCommentedCode(trivia, previousTrivia)) {
-        reportIssue(trivia.line());
-      }
+    super.visitTrivia(trivia);
+  }
 
-    } else if (!isPHPDoc(trivia)) {
-      String[] lines = regexpToDivideStringByLine.split(getContents(trivia.text()));
+  private void collectSingleLineComment(SyntaxTrivia trivia) {
+    // summarise previously collected single line comments if they have formed a consistent block.
+    if (!singleLineTrivias.isEmpty() && isNewCommentBlock(trivia)) {
+      checkSingleLineComments();
+    }
 
-      for (int lineOffset = 0; lineOffset < lines.length; lineOffset++) {
-        if (isCommentedCode(lines[lineOffset])) {
-          reportIssue(trivia.line() + lineOffset);
-          break;
-        }
-      }
+    // report valid syntax if it is the first line of a possible consistent block of single line comments
+    if (singleLineTrivias.isEmpty() && isParsableCode(trivia.text().replaceAll(SINGLE_LINE_COMMENT_REPLACE, " "))) {
+      context().newIssue(this, trivia, MESSAGE);
+    } else {
+      singleLineTrivias.addLast(trivia);
```

Comment:
I'm not sure if it makes sense do the `singleLineTrivias.addLast(trivia);` . Currently, if you have a block of code commented out by single-line comments:
```php
// echo $x;
// echo $y;
// echo $z;
```

An issue will be raised for each individual line. Maybe this is too noisy.